src/batched/KokkosBatched_SolveUTV_TeamVector_Internal.hpp

```diff
@@ -142,39 +143,45 @@ namespace KokkosBatched {
 		   one,
 		   T, ts0, ts1,
 		   W, ws0, ws1);
+	member.team_barrier();
 	
 	/// X = V^T W
 	TeamVectorGemmInternal<Algo::Gemm::Unblocked>
 	  ::invoke(member,
-		   m, nrhs, matrix_rank, 
+		   n, nrhs, matrix_rank, 
 		   one,
 		   V, vs1, vs0,
 		   W, ws0, ws1,
 		   zero,
 		   X, xs0, xs1);
+	member.team_barrier();
```

Comment:
No barriers or reductions at the end of Gemm, so team_barrier() aded since X is immediately reused in pivot reorder that follows.